src/coreclr/src/jit/rationalize.cpp

```diff
@@ -776,9 +759,20 @@ Compiler::fgWalkResult Rationalizer::RewriteNode(GenTree** useEdge, Compiler::Ge
             {
                 // Rewrite this as an explicit load.
                 JITDUMP("Rewriting GT_SIMD array init as an explicit load:\n");
-                unsigned int baseTypeSize = genTypeSize(simdNode->gtSIMDBaseType);
-                GenTree*     address = new (comp, GT_LEA) GenTreeAddrMode(TYP_BYREF, simdNode->gtOp1, simdNode->gtOp2,
-                                                                      baseTypeSize, OFFSETOF__CORINFO_Array__data);
+
+                GenTree* address;
+                if (simdNode->IsBinary())
+                {
+                    address = new (comp, GT_LEA)
+                        GenTreeAddrMode(TYP_BYREF, simdNode->GetOp(0), simdNode->GetOp(1),
+                                        genTypeSize(simdNode->gtSIMDBaseType), OFFSETOF__CORINFO_Array__data);
+                }
+                else
+                {
+                    assert(simdNode->IsUnary());
+                    address = new (comp, GT_LEA)
+                        GenTreeAddrMode(TYP_BYREF, simdNode->GetOp(0), nullptr, 0, OFFSETOF__CORINFO_Array__data);
+                }
```

Comment:
I would probably have structured this so that only the index is determined by the if/else, and then retain the common code for creating `address`. Otherwise the reader has to look more closely to understand that's the only difference.

Reply:
Sounds good to me. I was mostly trying to avoid setting LEA's scale in the base only case.

Reply:
On a side note - I wonder why it this done in rationalization. Seems like a lowering thing. Or even an import thing.

Reply:
Once upon a time we had a plan to move addressing modes earlier in the JIT (hence putting this in raionalization which is theoretically designed to go away once we have rational IR throughout). The idea would be that the creation of addressing modes would be a target-dependent thing, but they would be created early so that we wouldn't have to mark nodes as `GTF_DONT_CSE` because we think they'll be part of an addressing mode.

Reply:
Sorry, my comment wasn't clear. I was talking about the transformation from GT_SIMD+SIMDIntrinsicInitArray to GT_IND, it seems like this could be done at import time. In fact, it seems that there may be a bug around this, SIMDIntrinsicInitArray is a load and doesn't have GTF_GLOB_REF set on it. Interesting, I thought that only HWINTRINSC has issues with "hidden" load/stores.

Reply:
Ah, thanks for the clarification. It would definitely make sense to do that earlier.